src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -222,7 +222,10 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     // Store original constants' offset for serialization purposes
     store_const_offsets(model);
 
-    auto partitioning = getPartitioning(model, m_cfg);
+    // Identify based on compiler version, user config and pattern
+    auto use_host_gather_quant = should_use_quantized_host_gather(model, npuw_props);
+
+    auto partitioning = getPartitioning(model, m_cfg, use_host_gather_quant);
```

Comment:
as the `should_use_quantized_host_gather()` looks way better than ever, and you have just one property to control the behavior, I think _now_ it is legit to update the `m_cfg` based on that like

```suggestion
    // Identify based on compiler version, user config and pattern
    if (should_use_quantized_host_gather(model, npuw_props)) {
        m_cfg.update({npuw::partitioning::host_gather.name(), QUANT);
    }
    auto partitioning = getPartitioning(model, m_cfg);
```

and let partitioning handle it this way via `m_cfg`.

Reply:
Use ctx

Reply:
`host_gather` is kept `bool`. As discussed, using context now